enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,460 @@
+
+- Discovery: The user can choose to run discovery if they want to understand what devices are available across the nodes in the cluster.
+    - ##### NOTE: This step is optional for users directly creating CRs. They might already know what CRs they need for step 2.
+- PV Creation: After the user has decided how to created PVs, they can choose to create PVs with either the LocalVolume CR or the AutoDetectVolume CR.
+
+
+## Design Details for `Auto discovery of local devices`
+
+API scheme for `LocalVolumeDiscovery` and `LocalVolumeNodeDiscovery` CR:
+
+```go
+type LocalVolumeDiscoveryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeDiscovery `json:"items"`
+}
+
+type LocalVolumeDiscovery struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              LocalVolumeDiscoverySpec   `json:"spec"`
+	Status            LocalVolumeDiscoveryStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeDiscoverySpec struct {
+	// Nodes on which the automatic detection policies must run.
+	// +optional
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	// If specified tolerations is the list of toleration that is passed to the
+	// LocalVolumeDiscovery Daemon
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type LocalVolumeDiscoveryStatus struct {
+	Phase string `json:"phase,omitempty"`
+}
+
+type LocalVolumeNodeDiscoveryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeNodeDiscovery `json:"items"`
+}
+
+type LocalVolumeNodeDiscovery struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              LocalVolumeNodeDiscoverySpec   `json:"spec"`
+	Status            LocalVolumeNodeDiscoveryStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeNodeDiscoverySpec struct {
+	NodeName string `json:"nodeName"`
+	// If specified tolerations is the list of toleration that is passed to the
+	// LocalVolumeDiscovery Daemon
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type LocalVolumeNodeDiscoveryStatus struct {
+	Phase      string `json:"phase,omitempty"`
+	TimeStramp string `json:"timestramp,omitempty"`
+	// Blockdevices will be containing the output of following command unmarshalled into the structured
+	// "lsblk -o name,model,vendor,kname,pkname,type,uuid,tran,ro,rm,parttype,serial,size,rota,mountpoint"
+	Blockdevices []Device `json:"blockdevices"`
+}
+
+type Device struct {
+	Name       string   `json:"name,omitempty"`
+	Model      string   `json:"model,omitempty"`
+	Vendor     string   `json:"vendor,omitempty"`
+	Kname      string   `json:"kname,omitempty"`
+	Pkname     string   `json:"pkname,omitempty"`
+	Type       string   `json:"type,omitempty"`
+	UUID       string   `json:"uuid,omitempty"`
+	Tran       string   `json:"tran,omitempty"`
```

Comment:
Please use proper names - what's `Rm`, `Tran`, `Ro` etc.

Reply:
Ahh `Rm,Tran,Ro` are the names already present in the lsblk properties.

Reply:
Yeah, but in the API I'd like to be more explicit, so `Removabe bool`, `TransportType <some enum>?`, `ReadOnly bool`, `KernelPath string`, etc.

I'd suggest to cut down the number of fields too.

Reply:
okay